Modules/xSharePoint/DSCResources/MSFT_xSPWebAppPolicy/MSFT_xSPWebAppPolicy.psm1

```diff
@@ -19,32 +44,32 @@ function Get-TargetResource
         $wa = Get-SPWebApplication -Identity $params.WebAppUrl -ErrorAction SilentlyContinue
 
         if ($null -eq $wa) { return $null }
-        
-        if ($wa.Policies.UserName -contains $params.UserName) {
-            $policyObject = $wa.Policies | Where-Object { $_.UserName -eq $params.UserName }
-            return @{
+
+        $members = @()
+        foreach ($policy in $wa.Policies) {
+            $member = @{}
+            $member.Username = $policy.UserName
+            $member.PermissionLevel = $policy.PolicyRoleBindings.Name
+            $member.ActAsSystemAccount = $policy.IsSystemUser
+            $members += $member
+        }
+
+        $returnval = @{
                 WebAppUrl = $params.WebAppUrl
-                UserName = $params.UserName
-                PermissionLevel = ($policyObject.PolicyRoleBindings | Select-Object -First 1).Name
-                ActAsSystemUser = $policyObject.IsSystemUser
+                Members = $members
+                MembersToInclude = $params.MembersToInclude
+                MembersToExclude = $params.MembersToExclude
                 InstallAccount = $params.InstallAccount
-            }
-        } else {
-            foreach($userName in $wa.Policies.UserName) {
-                $claimsPrincipal = New-SPClaimsPrincipal -EncodedClaim $userName -ErrorAction SilentlyContinue
-                if (($null -ne $claimsPrincipal) -and ($claimsPrincipal.Value -eq $params.UserName)) {
-                    $policyObject = $wa.Policies | Where-Object { $_.UserName -eq $userName }
-                    return @{
-                        WebAppUrl = $params.WebAppUrl
-                        UserName = $claimsPrincipal.Value
-                        PermissionLevel = ($policyObject.PolicyRoleBindings | Select-Object -First 1).Name
-                        ActAsSystemUser = $policyObject.IsSystemUser
-                        InstallAccount = $params.InstallAccount
-                    }
-                }
-            }
-            return $null 
         }
+
+        if ($wa.Properties.ContainsKey("portalsuperuseraccount")) {
+            $returnval.SuperUserAccount = $wa.Properties["portalsuperuseraccount"]
+        }
+        if ($wa.Properties.ContainsKey("portalsuperreaderaccount")) {
+            $returnval.SuperReaderAccount = $wa.Properties["portalsuperreaderaccount"]
+        }
```

Comment:
Why is this necessary? Is this a requirement from DSC or something we specified?

The reason I did this is that I need these values in the Test-TargetResource and wanted to prevent another call to Invoke-xSharePointCommand in that method. Since I was already requesting information from the Get method, I included these parameters into the results set.